crates/bevy_ecs/src/query/iter.rs

```diff
@@ -397,8 +316,13 @@ where
         QueryIterationCursor {
             fetch,
             filter,
-            table_id_iter: query_state.matched_table_ids.iter(),
-            archetype_id_iter: query_state.matched_archetype_ids.iter(),
+            // SAFE: The associated IDs are exactly the same size and alignment and
```

Comment:
Technically this relies on the unspecified layouts of `TableId`, `ArchetypeId`, and `slice::Iter`, which are all [`repr(Rust)`](https://doc.rust-lang.org/nomicon/transmutes.html) types.
By adding `#[repr(transparent)]` to `TableId` and `ArchetypeId`, it becomes safe to `unsafe{ std::slice::from_raw_parts(_.as_ptr() as *const usize, _.len()) }.iter()` IIUC.

Reply:
Undoing this because `std::slice::Iter` is  `repr(Rust)`.